### PR TITLE
Add Ability to choose what resource is mined

### PR DESCRIPTION
**Pull request by eun4786**
https://feathub.com/Anuken/Mindustry/+243
ore selection
When you click on a drill, the resource table is displayed and mine the selected resources from the table.

### CODE REVIEW

**Comment by Quezler** [comment]
+1

**Comment by Kieaer** [comment]
+1

**Comment by Anuken** [comment]
I don't agree with making this configurable. Even if it's hidden behind a menu, this adds unnecessary UI and complexity that isn't needed in most cases.

**Comment by DeltaNedas** [comment]
why not a drill setting tool that sets target resource for any drill you click instantly, no bloat ui for drills just one thing

File: core/src/io/anuke/mindustry/world/blocks/ItemSelection.java [hunk 1]
```diff
@@ -40,4 +40,29 @@ public static void buildItemTable(Table table, Prov<Item> holder, Cons<Item> con
 
         table.add(cont);
     }
+
+    public static void buildDrillItemTable(Table table, Array<Item> items, Prov<Item> holder, Cons<Item> consumer){
+
```
Comment (YaFou) [inline core/src/io/anuke/mindustry/world/blocks/ItemSelection.java]: Why there is a line break? There is no need.

Comment (Arkanic) [reply core/src/io/anuke/mindustry/world/blocks/ItemSelection.java]: try importing the mindustry code format
~~i've done the same too many times~~

Comment (YaFou) [reply core/src/io/anuke/mindustry/world/blocks/ItemSelection.java]: Mindustry IntelliJ code format: https://github.com/Anuken/Mindustry/blob/master/.github/Mindustry-CodeStyle-IJ.xml

**Comment by Jaeiya** [comment]
> I don't agree with making this configurable. Even if it's hidden behind a menu, this adds unnecessary UI and complexity that isn't needed in most cases.

I've played several games where I wanted to max out production efficiency of a specific resource but it was too close to another (sand, scrap, coal, etc...). I agree it's a QoL feature that may not be necessary, but there is definitely a precedence for it.

**Comment by XEN0PHIL** [comment]
What about making it drill everything by default?
For example there's 3 lead and 1 copper in a 2x2 square, then mechanical drill will output 75% lead, 25% copper.

**Comment by Jaeiya** [comment]
@XEN0PHIL But then you'd have to use a sorter and that adds unnecessary complexity to your lines. I find in most cases, if there's a resource bleeding into another, I only want one of them; I very rarely if ever have found myself needing both resources equally.

**Comment by YaFou** [comment]
I manage easily my drills with multiples ores below. And often, the ore is far from others, so we don't need to select the resource which we want to mine.
**Also, this isn't necessary but it's can be a good extra feature as an option in settings.**

**Comment by matkoniecz** [comment]
"extra feature as an option in settings" are making testing and avoiding bugs significantly harder.

**Comment by CA6** [comment]
eh this is QoL stuff that would be nice but I'd like like how delta said it a singular menu in the build menu to save ui computing

**Comment by Anuken** [comment]
This PR can still be used for discussion, but I will not be merging it in its current form.

**Pull request closed without merging**